### PR TITLE
ci: add ignored paths & skip running Buildkite tests when git diff is empty

### PR DESCRIPTION
**Pull request by Jolg42**
It breaks the infinite loop of releases and makes our Buildkite setup better by avoiding testing and building engines when it’s not needed.

See example run
https://buildkite.com/prisma/test-prisma-engines/builds/21127#018b1e2a-d699-4d44-980e-90b8a738c8e0/130-131

Screenshot 2023-10-11 at 12 04 13

### CODE REVIEW

**Comment by Jolg42** [comment]
A long long time ago in prisma/prisma repo we had this https://github.com/prisma/prisma/blob/2.10.x/src/.buildkite/test/buildkite-entry.sh though I like the git diff approach better because it has no dependencies.

File: .buildkite/engineer [hunk 1]
```diff
@@ -1,5 +1,32 @@
...
+    # So we do not start Buildkite
+    if [ -z "${GIT_DIFF}" ]; then
+        echo "No changes found for the previous commit in paths that are not ignored, this run will now be skipped."
+        exit 0
+    else
+        # Note that printf works better for displaying line returns in CI
+        printf "Changes found for the previous commit in paths that are not ignored: \n\n${GIT_DIFF}\n\nThis run will continue...\n"
+    fi
+else
+    echo "We are in the build pipeline."
```
Comment (miguelff) [inline .buildkite/engineer:26]: Is there a real intention to print this message when building? Or is it a leftover?

**Review by miguelff — APPROVED** [review]
Not blocking feedback provided.

File: .buildkite/engineer (hunk 1 above)
Comment (Jolg42) [reply .buildkite/engineer]: It's to be clear/explicit about what's happening.
I made it more generic now
